ui/app/src/modules/Preview/ToolBar.tsx

```diff
@@ -290,80 +192,11 @@ export function AddressBar(props: AddressBarProps) {
       <IconButton className={classes.iconButton} aria-label="search" onClick={handleClick}>
         <MoreVertRounded/>
       </IconButton>
-      <Menu
+      <ComponentMenu
         anchorEl={anchorEl}
-        open={Boolean(anchorEl)}
-        onClose={handleClose}
-      >
-        <MenuItem onClick={() => handleEdit('form')}>
-          <FormattedMessage
-            id="previewToolBar.menu.edit"
-            defaultMessage="Edit"
-          />
-        </MenuItem>
-        {
-          (!publishDialog.item?.lockOwner && !publishDialog.item?.isLive) &&
-          <MenuItem onClick={() => handleEdit('schedule')}>
-            <FormattedMessage
-              id="previewToolBar.menu.schedule"
-              defaultMessage="Schedule"
-            />
-          </MenuItem>
-        }
-        {
-          (!publishDialog.item?.lockOwner && !publishDialog.item?.isLive) &&
-          <MenuItem onClick={() => handleEdit('publish')}>
-            <FormattedMessage
-              id="previewToolBar.menu.publish"
-              defaultMessage="Publish"
-            />
-          </MenuItem>
-        }
-        <MenuItem onClick={handleClose}>
-          <FormattedMessage
-            id="previewToolBar.menu.history"
-            defaultMessage="History"
-          />
-        </MenuItem>
-        <MenuItem onClick={handleClose}>
-          <FormattedMessage
-            id="previewToolBar.menu.dependencies"
-            defaultMessage="Dependencies"
-          />
-        </MenuItem>
-        <MenuItem onClick={handleClose}>
-          <FormattedMessage
-            id="previewToolBar.menu.delete"
-            defaultMessage="Delete"
-          />
-        </MenuItem>
-        <MenuItem onClick={handleClose} className={classes.separator}>
-          <FormattedMessage
-            id="previewToolBar.menu.infoSheet"
-            defaultMessage="Info Sheet"
-          />
-        </MenuItem>
-        <MenuItem onClick={() => handleEdit('template')}>
-          <FormattedMessage
-            id="previewToolBar.menu.editTemplate"
-            defaultMessage="Edit Template"
-          />
-        </MenuItem>
-        <MenuItem onClick={() => handleEdit('controller')}>
-          <FormattedMessage
-            id="previewToolBar.menu.editController"
-            defaultMessage="Edit Controller"
-          />
-        </MenuItem>
-      </Menu>
-      {
-        dialogConfig.open &&
-        <EmbeddedLegacyEditors dialogConfig={dialogConfig} setDialogConfig={setDialogConfig} getPath={getPath}/>
-      }
-      {
-        publishDialog.open &&
-        <PublishDialog scheduling={publishDialog.scheduling} items={[publishDialog.item]} onClose={onClosePublish}/>
-      }
+        setAnchorEl={setAnchorEl}
+        site={site}
+      />
```

Comment:
- Why does the ComponentMenu need the ability to **set** the anchor?
- Follow same API as material components (use both anchor and open props, onClose, etc)